const-prop: Recognize Python's "..." * N expressions as constant

IagoAbal opened the pull request:
In Python, `"a" * 3` evaluates to `"aaa"`. We will not evaluate these expressions but we will at least recognize them as (arbitrary) string constants so they can be matched by `"..."` patterns. We don't check whether the N is indeed an integer, but this should be fine.

Closes FLOW-75

test plan:
make test # two new tests

aryx reviewed (approved):
A test would be nice.
Also in theory we should also update Eval_generic.ml

IagoAbal commented:
> A test would be nice.

Whoops, I have the tests but forgot to commit them!

> Also in theory we should also update Eval_generic.ml

We're not really evaluating `"a" * 3` but just recognizing it as an "arbitrary string constant".

IagoAbal committed:
const-prop: Recognize Python's "..." * N expressions as constant

In Python, `"a" * 3` evaluates to `"aaa"`. We don't check whether the N
is indeed an integer, but this should be fine.

Closes FLOW-75

test plan:
make test # two new tests
